sherpa/astro/ui/utils.py

```diff
@@ -12364,7 +12367,7 @@ def calc_kcorr(self, z, obslo, obshi, restlo=None, resthi=None,
     # Session Text Save Function
     ###########################################################################
 
-    def save_all(self, outfile=None, clobber=False):
+    def save_all(self, outfile=None, clobber=False, outfh=None):
```

Comment:
I guess you'd do it as "check for `None`, then check for a file handle, otherwise assume a string (I'm a little wary of checking for strings, since it has bitten us in the past when we've not done it correctly - e.g. np.string_).

Overloading outfile is in many clearer for the code, and is something we do a lot of, so it wouldn't be a terrible choice. I think I wanted to separate out the two here just to be a bit cleaner/clearer in the interface, but would be happy either for

* the string buffer support to be dropped for now

* your suggestion be used (i.e. drop the `outfh` argument, which isn't a very useful name)